subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -0,0 +1,737 @@
+
+		/* AuxPtr */
+		if (_hp.aux_ptr) {
+			_pp -= sizeof(struct ext_adv_aux_ptr);
+		}
+		{
+			struct ext_adv_aux_ptr *aux;
+
+			pp -= sizeof(struct ext_adv_aux_ptr);
+
+			/* NOTE: Aux Offset will be set in advertiser LLL event
+			 */
+			aux = (void *)pp;
+			aux->chan_idx = 0; /* FIXME: implementation defined */
+			aux->ca = 0; /* FIXME: implementation defined */
+			aux->offs_units = 0; /* FIXME: implementation defined */
+			aux->phy = find_lsb_set(adv->lll.phy_s) - 1;
+		}
+
+		/* TODO: reduce duplicate code if below remains similar to
+		 * primary PDU
+		 */
+		if (_hs.aux_ptr) {
+			struct ext_adv_aux_ptr *aux;
+
+			_ps -= sizeof(struct ext_adv_aux_ptr);
+			ps -= sizeof(struct ext_adv_aux_ptr);
+
+			/* NOTE: Aux Offset will be set in advertiser LLL event
+			 */
+			aux = (void *)ps;
+			aux->chan_idx = 0; /* FIXME: implementation defined */
+			aux->ca = 0; /* FIXME: implementation defined */
+			aux->offs_units = 0; /* FIXME: implementation defined */
+			aux->phy = find_lsb_set(adv->lll.phy_s) - 1;
+		}
+
+		/* ADI */
+		{
+			struct ext_adv_adi *ap, *as;
+			u16_t did = UINT16_MAX;
+
+			pp -= sizeof(struct ext_adv_adi);
+			ps -= sizeof(struct ext_adv_adi);
+
+			ap = (void *)pp;
+			as = (void *)ps;
+
+			if (_hp.adi) {
+				struct ext_adv_adi *_adi;
+
+				_pp -= sizeof(struct ext_adv_adi);
+				_ps -= sizeof(struct ext_adv_adi);
+
+				/* NOTE: memcpy shall handle overlapping buffers
+				 */
+				memcpy(pp, _pp, sizeof(struct ext_adv_adi));
+				memcpy(ps, _ps, sizeof(struct ext_adv_adi));
+
+				_adi = (void *)_pp;
+				did = _adi->did;
+			} else {
+				ap->sid = adv->sid;
+				as->sid = adv->sid;
+			}
+
+			did++;
+
+			ap->did = did;
+			as->did = did;
+		}
+
+		/* No CTEInfo field in primary channel PDU */
+
+		/* No TargetA non-conn non-scan advertising  */
+
+		/* No AdvA in primary channel due to AuxPtr being added */
+
+		/* NOTE: AdvA in aux channel is also filled at enable and RPA
+		 * timeout
+		 */
+		if (hs->adv_addr) {
+			void *bdaddr;
+
+			if (_hs.adv_addr) {
+				_ps -= BDADDR_SIZE;
+				bdaddr = _ps;
+			} else {
+				_pp -= BDADDR_SIZE;
+				bdaddr = _pp;
+			}
+
+			ps -= BDADDR_SIZE;
+
+			memcpy(ps, bdaddr, BDADDR_SIZE);
+		}
+
+		lll_adv_aux_data_enqueue(lll_aux, is);
+		lll_adv_data_enqueue(lll, ip);
+	} else {
+		sync = (void *)HDR_LLL2EVT(lll_sync);
+	}
+
+	sync->interval = interval;
+
+	pdu = lll_adv_sync_data_peek(lll_sync);
+	pdu->type = PDU_ADV_TYPE_AUX_SYNC_IND;
+	pdu->rfu = 0U;
+	pdu->chan_sel = 0U;
+	pdu->tx_addr = 0U;
+	pdu->rx_addr = 0U;
+
+	t = (void *)&pdu->adv_ext_ind;
+	ht = (void *)t->ext_hdr_adi_adv_data;
+	pt = (u8_t *)ht + sizeof(*ht);
+	*(u8_t *)&_ht = *(u8_t *)ht;
+	*(u8_t *)ht = 0;
+	_pt = pt;
+
+	/* Non-connectable and Non-scannable adv mode */
+	t->adv_mode = 0;
+
+	/* No AdvA */
+	/* No TargetA */
+
+	/* TODO: CTEInfo */
+
+	/* No ADI */
+
+	/* TODO: AuxPtr */
+
+	/* No SyncInfo */
+
+	/* TODO: TxPower */
+	if (flags & BIT(6)) {
```

Comment:
Please use a define/enum instead of `BIT(6)`

Reply:
addressed.